engine/src/main/java/org/hibernate/search/engine/mapper/mapping/building/impl/IndexSchemaFilter.java

```diff
@@ -222,21 +233,69 @@ public IndexSchemaFilter compose(IndexedEmbeddedDefinition definition, IndexedEm
 			cyclicRecursionPath += definition.relativePrefix();
 			throw log.indexedEmbeddedCyclicRecursion( cyclicRecursionPath, definition.definingTypeModel() );
 		}
+		Set<String> conflictingPaths = findConflictingIncludeExcludePaths( definition );
+		if ( !conflictingPaths.isEmpty() ) {
+			throw log.conflictingIncludeExcludePaths(
+					definition.includePaths(), definition.excludePaths(), conflictingPaths );
+		}
 
 		// The new depth filter according to the given includeDepth
 		DepthFilter newDepthFilter = DepthFilter.of( definition.includeDepth() );
 
 		// The new path filter according to the given includedPaths
-		PathFilter newPathFilter = PathFilter.of( definition.includePaths() );
+		PathFilter newPathFilter = PathFilter.of( definition.includePaths(), definition.excludePaths() );
 
 		return new IndexSchemaFilter(
 				this, definition, pathTracker,
 				newDepthFilter, newPathFilter
 		);
 	}
 
+	private Set<String> findConflictingIncludeExcludePaths(IndexedEmbeddedDefinition definition) {
+		Set<String> conflictingPaths = new LinkedHashSet<>();
+		for ( String path : definition.excludePaths() ) {
+			conflictingPathExplicitlyIncluded( definition.relativePrefix() + path )
+					.ifPresent( conflictingPaths::add );
```

Comment:
I wonder what's the point of this...

We're testing a path that's explicitly excluded. Since a given filter cannot use both `includePaths` and `excludePaths`, it's safe to say that paths won't be explicitly included by the same filter.

As to parent filters... I would expect path tracking to be enough to detect such conflicts, i.e. the parent filter will have an `excludePaths` filter for a path that is simply never encountered, so you'll get an error automatically?